src/semantics/Action.re

```diff
@@ -3408,6 +3420,47 @@ let rec syn_perform_block =
     let (ze, u_gen) = ZExp.new_EmptyHole(u_gen);
     let zblock = ZExp.BlockZE(lines, ze);
     syn_perform_block(~ci, ctx, keyword_action(k), (zblock, Hole, u_gen));
+
+  /* Variable Splitting */
+
+  | (Construct(SOp(op)), BlockZE(lines, zexp))
+      when zexp_is_suitable_for_var_split(zexp) =>
+    let split_var = (pos, err, verr, name) => {
+      let left_var = String.sub(name, 0, pos);
+      let right_var = String.sub(name, pos, String.length(name) - pos);
+      let cursor =
+        ZExp.CursorE(
+          OnText(String.length(name) - pos),
+          Var(err, verr, right_var),
+        );
```

Comment:
Here you're placing the cursor on the right split var. Let's instead place the cursor to the right of the new op. This will be necessary in the long run for when we have multi-character ops (e.g. did the user mean a `+` or a `++`?).

Reply:
If the op is a space, though, you will want to place the cursor to the left of the right split var